src/charm.py

```diff
@@ -1167,6 +1211,35 @@ def generate_auth_env_vars(self, conf: Dict[str, Dict[str, Any]]) -> Dict[str, s
             env_vars[auth_var_prefix + str(var).upper()] = str(conf[auth_mode][var])
         return env_vars
 
+    @property
+    def _client_config(self) -> ClientConfig:
```

Comment:
```suggestion
    def _oauth_client_config(self) -> ClientConfig:
```
It would be nice to have this hidden away somewhere, but if it's going to be in the main charm codebase, it may as well be explicit about the fact that this property is strictly for oauth.